src/main/java/io/stargate/sgv2/jsonapi/service/projection/ProjectionPath.java

```diff
@@ -0,0 +1,152 @@
+package io.stargate.sgv2.jsonapi.service.projection;
+
+import io.stargate.sgv2.jsonapi.exception.ProjectionException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * An immutable representation of a projection path composed of segments.
+ *
+ * <p>A projection path is created from an encoded string (e.g., "pricing.price&amp;.usd"), where
+ * segments are separated by dots ('.') and the escape character ('&amp;') allows literal dots or
+ * ampersands within segments.
+ *
+ * <p>Use {@link #from(String)} to create an instance, {@link #getSegments()} to access the
+ * segments, and {@link #encode()} to get the dot-separated path string.
+ */
+public final class ProjectionPath {
+  private final List<String> segments;
+
+  private ProjectionPath(List<String> segments) {
+    // Make a defensive copy to preserve immutability.
```

Comment:
One benefit of removing accessor to segments is that we can avoid creating defensive copy.